src/tracker.cpp

```diff
@@ -92,7 +92,10 @@ void Tracker::set_subject(Subject *subject) {
 
 // --- Event Tracking
 
-void Tracker::track(Payload payload, const string &event_id, vector<SelfDescribingJson> &contexts) {
+vector<string> Tracker::track(const Event &event) {
```

Comment:
I'm in doubt if we should be consistent with the Java Tracker about the list of the `event_id` we return. The reason behind this weird choice is only imputable to the eCommerce transaction event and items, which are all events, but some of them are used as entities. I think we should deprecate them and provide a better alternative.
Considering you don't have them here, we could just return a single event_id that makes more sense and think how to deal with this problem in the trackers that support those events. Unfortunately, that was a bad design decision that we continue to bring forward because customers have used those.
Yet another point of discussion. I let you decide what's the best for now.

Reply:
Ecommerce events aside, the only reason I can think of to potentially return multiple `event_id`s would be if the payload could be split in two if it was too big? Any others?

This is a long way from happening, so I agree it's better to return just the string. In the Java tracker docs I warned about future deprecation of eCommerce events, though we didn't talk yet about an actual timeline for doing so.

Reply:
I'm more than happy to change this to return a single event ID, it's more understandable and easier for users to handle a single event ID in response instead of a list. I was just overly cautious. I agree that if it's just the ecommerce edge case and we don't expect to have more such cases in the future, it is not necessary. Changed it to return a string.